src/tags_int.cpp

```diff
@@ -2577,26 +2577,69 @@ std::ostream& printExifUnit(std::ostream& os, const Value& value, const ExifData
 
 std::ostream& printLensSpecification(std::ostream& os, const Value& value, const ExifData*) {
   std::ios::fmtflags f(os.flags());
-  if (value.count() != 4 || value.toRational(0).second == 0 || value.toRational(1).second == 0) {
+  // check type and count of values
+  if (value.typeId() != unsignedRational || value.count() != 4 ||
+      // divisor may be zero only if dividend is not zero
+      value.toRational(0).first != 0 && value.toRational(0).second == 0 ||
+      value.toRational(1).first != 0 && value.toRational(1).second == 0 ||
+      value.toRational(2).first != 0 && value.toRational(2).second == 0 ||
+      value.toRational(3).first != 0 && value.toRational(3).second == 0) {
     os << "(" << value << ")";
     return os;
   }
-  const int64_t len1 = value.toInt64(0);
-  const int64_t len2 = value.toInt64(1);
-
-  auto [r1, s1] = value.toRational(2);
-  auto [r2, s2] = value.toRational(3);
-  os << len1;
-  if (len2 != len1) {
-    os << "-" << len2;
+  // values numerically are ok, so they can be converted
+  // here first and second can be zero, so initialise float with 0.0
+  float focalLength1 = 0.0;
```

Comment:
[suggestion] Some compilers at certain warning levels would complain about not using the same types here (float variable vs double literals). Could you replace those `0.0` with `0.0F` when comparing or assigning float variables?